tests/transforms/test_split_non_commuting.py

```diff
@@ -293,6 +293,28 @@ def test_mixed_wires_obs_measurement_types(self, meas_type_1, meas_type_2):
         assert qml.equal(split[1].measurements[0], meas_type_2())
         assert qml.equal(split[1].measurements[1], meas_type_2(wires=[0, 1]))
 
+    def test_batch_of_tapes(self):
+        """Test that `split_non_commuting` can transform a batch of tapes"""
+
+        # create a batch with two fictitious tapes
+        tape1 = qml.tape.QuantumScript(
+            [qml.RX(1.2, 0)], [qml.expval(qml.X(0)), qml.expval(qml.Y(0)), qml.expval(qml.X(1))]
+        )
+        tape2 = qml.tape.QuantumScript(
+            [qml.RY(0.5, 0)], [qml.expval(qml.Z(0)), qml.expval(qml.Y(0))]
+        )
+        batch = [tape1, tape2]
+
+        # test transform on the batch
+        new_batch, _ = split_non_commuting(batch)
+
+        # test that result is equal to the one obtained with `map_batch_transform`
+        new_batch_check, _ = qml.transforms.map_batch_transform(split_non_commuting, batch)
```

Comment:
I'm aiming to be able to replace `map_batch_transform` and deprecate it.  Luckily in this case, we know exactly what the output should be.  The first tape should split in two, one measuring `[qml.expval(qml.X(0)), qml.expval(qml.X(1))]` and the second measuring `[qml.expval(qml.Z(0)), qml.expval(qml.Y(0))]`.

We should also test function too.

Since the output for `split_non_commuting` is just reshuffling, one thing I like to do in this case is use strings in place of the result values, so we can focus on their reshuffling. Something like:

```
result = ("a", "b", "c", "d")
assert fn(result) == [("a", "b"), ("c", d")]
```

Reply:
That'd be great thanks.  We want these tests to still be valid if we deprecate `map_batch_transform`.